### PR TITLE
LUCENE-8626: Standardize Lucene Test Files

### PR DESCRIPTION
**Pull request by MarcusSorealheis**
# Description

Janitor here, moving a closer to a standard test file naming convention, starting with Lucene. 

# Solution

Janitorial work based on the work that @cpoerschke started a couple years ago in the ticket from the title.Almost every change was identical. More to come. PR was already growing too big.

# Tests

These are all tests.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Comment by ErickErickson** [comment]
LGTM. We can check these in piecemeal rather than do them all at once, as you say the PRs get massive...

**Comment by dweiss** [comment]
Wouldn't it be better to add test convention enforcement first, followed by gradual rename of all convention-excluded files? I filed a suggestion on how this could be done - it doesn't have to be that particular one; can be anything but would be better than nothing?

**Comment by MarcusSorealheis** [comment]
@dweiss I'm of the opinion that they could happen in parallel because we know in which direction we are going given the current state. The enforcement will obviously be a plus, but both the enforcement and the renaming are both in flight and gradually landing.

**Comment by dweiss** [comment]
ok.

File: lucene/spatial-extras/src/test/org/apache/lucene/spatial/TestQueryEqualsHashCode.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import org.apache.lucene.spatial.query.SpatialArgs;
+import org.apache.lucene.spatial.query.SpatialOperation;
+import org.apache.lucene.spatial.serialized.SerializedDVStrategy;
+import org.apache.lucene.spatial.vector.PointVectorStrategy;
+import org.apache.lucene.util.LuceneTestCase;
+import org.junit.Test;
+import org.locationtech.spatial4j.context.SpatialContext;
+import org.locationtech.spatial4j.shape.Shape;
+
+public class TestQueryEqualsHashCode extends LuceneTestCase {
```
Comment (anshumg) [inline lucene/spatial-extras/src/test/org/apache/lucene/spatial/TestQueryEqualsHashCode.java:38]: Did you intend to add this as a duplicate file?

Comment (MarcusSorealheis) [reply lucene/spatial-extras/src/test/org/apache/lucene/spatial/TestQueryEqualsHashCode.java:38]: what is the duplicate?

Comment (MarcusSorealheis) [reply lucene/spatial-extras/src/test/org/apache/lucene/spatial/TestQueryEqualsHashCode.java:38]: I intended to remove the previous file `lucene/spatial-extras/src/test/org/apache/lucene/spatial/QueryEqualsHashCodeTest.java`

Comment (MarcusSorealheis) [reply lucene/spatial-extras/src/test/org/apache/lucene/spatial/TestQueryEqualsHashCode.java:38]: I see, I didn't remove the original in that case good catch

Comment (MarcusSorealheis) [reply lucene/spatial-extras/src/test/org/apache/lucene/spatial/TestQueryEqualsHashCode.java:38]: that was a `git fart` on my end.

Comment (MarcusSorealheis) [reply lucene/spatial-extras/src/test/org/apache/lucene/spatial/TestQueryEqualsHashCode.java:38]: that's why I am keeping these small

**Review by CaoManhDat — APPROVED** [review]
LGTM!, Gonna commit it after waiting for a day.

**Comment by CaoManhDat** [comment]
Thank you @MarcusSorealheis

**Pull request merged**